### PR TITLE
Download logs org failure

### PR DESCRIPTION
**Pull request by davidmreed**
When a scratch org build fails, allow the user to download the log from the Toast message, and offer them the org id to seek support.

### CODE REVIEW

File: src/js/store/orgs/actions.ts [hunk 1]
```diff
@@ -182,11 +182,27 @@ export const provisionFailed =
...
           heading: msg,
-          details: message,
+          details: detailMsg,
           variant: 'error',
+          linkUrl: window.api_urls.scratch_org_log(model.id),
+          linkText: t('Download build log.'),
+          linkDownload: true,
+          linkDownloadFilename: t('Metecho Org {{orgId}} Build Log.txt', {
+            message,
+            orgId: model.id,
```
Comment (jstvz) [inline src/js/store/orgs/actions.ts:204]: What does `i18next.t` do when the key is missing? Fail silently?

File: docs/auth.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
+# Metecho Authentication
```
Comment (TheBitShepherd) [inline docs/auth.md:1]: This section is much need. Thanks for writing this up!

Comment (davidmreed) [reply docs/auth.md:1]: Oh jeez that was supposed to be in a different PR entirely. Retracting and pushing elsewhere...

**Review by jstvz — APPROVED** [review]
@davidmreed LGTM, but the code coverage threshold is 100% and it looks like you've got an uncovered branch somewhere blocking merge.

**Comment by davidmreed** [comment]
W-12727028

**Review by prescod — APPROVED** [review]

**Pull request merged**